ext/opcache/jit/zend_elf.c

```diff
@@ -46,7 +49,20 @@ void zend_elf_load_symbols(void)
 	zend_elf_header hdr;
 	zend_elf_sectheader sect;
 	int i;
+#if defined(__linux__)
 	int fd = open("/proc/self/exe", O_RDONLY);
+#elif defined(__FreeBSD__)
+	char path[PATH_MAX];
+	size_t pathlen = sizeof(path);
+	int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
+	if (sysctl(mib, 4, path, &pathlen, NULL, 0) == -1) {
+             return;
+	}
+	int fd = open(path, O_RDONLY);
+#elif defined(__APPLE__)
```

Comment:
This should be just `#else`, to account for unknown platforms.

Reply:
OK fair enough.